Add per-effect enable/disable toggles in BepInEx config (#12)
Brollyy/BopVisualEffects: README.md

See [this document](docs/effects/README.md) for the full list of effects this mod enables.

## Configuration

Each effect can be individually enabled or disabled in the BepInEx config file (`BepInEx/config/com.brollyy.bopvisualeffects.cfg`), under the `[Effects]` section.

> The config file is at `<GameRoot>/BepInEx/config/BopVisualEffects.cfg`. It's generated automatically on first load. To disable an effect, set its entry to `false`:

Reviewer: No, the Plugin GUID is `BopVisualEffects`, that's what should be used

> The config file is at `<GameRoot>/BepInEx/config/BopVisualEffects.cfg`. It's generated automatically on first load. To disable an effect, set its entry to `false`:

Reviewer: @copilot align these paths to both use the correct file name

See [this document](docs/effects/README.md) for the full list of effects this mod enables.

## Configuration

Each effect can be individually enabled or disabled in the BepInEx config file (`BepInEx/config/com.brollyy.bopvisualeffects.cfg`), under the `[Effects]` section.

> The config is generated automatically on first load. To disable an effect, set its entry to `false`:

Reviewer: Explain where they can find the config to do so.

```ini
[Effects]

## Whether the Camera Shake effect is active.
# Setting type: Boolean
# Default value: true
CameraShake.Enabled = false
```

> All available config keys:

Reviewer: These should be listed next to each effect in `docs/effects/README.md` instead. Here, only include a link to where they can find the config keys for effects

| Effect | Config Key |
|---|---|
| Camera Shake | `CameraShake.Enabled` |
| Camera Tilt | `CameraTilt.Enabled` |
| Color Tint | `ColorTint.Enabled` |
| Fog | `Fog.Enabled` |
| Letterbox | `Letterbox.Enabled` |
| Pixel Grid | `PixelGrid.Enabled` |
| Scanlines | `Scanlines.Enabled` |
| Screen Noise | `ScreenNoise.Enabled` |
| Vignette | `Vignette.Enabled` |
| Zoom In | `ZoomIn.Enabled` |
| Zoom Out | `ZoomOut.Enabled` |
| Zoom Pulse | `ZoomPulse.Enabled` |

> Disabled effects are removed from the editor template picker and will not run during playback.

Reviewer: ```suggestion
Disabled effects are removed from the in-game editor and will not run during playback.
```

Each effect can be individually enabled or disabled in the BepInEx config file (`BepInEx/config/com.brollyy.bopvisualeffects.cfg`), under the `[Effects]` section.

> The config file is located in your Bits & Bops game folder. It is generated automatically on first load. To disable an effect, set its entry to `false`:

Reviewer: @copilot The location has to be more precise - it's a config file generated by BepInEx, provide specific path relative to the game folder

> The config file is at `<GameRoot>/BepInEx/config/com.brollyy.bopvisualeffects.cfg`. It is generated automatically on first load. To disable an effect, set its entry to `false`:

Reviewer: ```suggestion
The config file is at `<GameRoot>/BepInEx/config/BopVisualEffects.cfg`. It's generated automatically on first load. To disable an effect, set its entry to `false`:

### Adding a new effect

1. Create a class in `BopVisualEffects/Effects/<EffectName>/` that implements `IVisualEffectDefinition`.

> 2. Implement `Id` (lowercase, stable identifier), `DisplayName` (human-readable), `ConfigKey` (PascalCase, used for config file entries), and `Description`.

Reviewer: ```suggestion
2. Fill in `Id` (lowercase, with spaces), `DisplayName` (human-readable), `ConfigKey` (PascalCase, used for config file entries) and `Description`.
```
